crates/holochain_persistence_lmdb/src/eav/lmdb.rs

```diff
@@ -78,18 +81,17 @@ where
         // use a clever key naming scheme to speed up exact match queries on the entity
         let mut new_eav = eav.clone();
         let mut key = format!("{}::{}", new_eav.entity(), new_eav.index());
-
         // need to check there isn't a duplicate key though and if there is create a new EAVI which
         // will have a more recent timestamp
         while let Ok(Some(_)) = self.lmdb.store.get(&reader, key.clone()) {
             new_eav = EntityAttributeValueIndex::new(&eav.entity(), &eav.attribute(), &eav.value())
-                .unwrap();
+                .map_err(|_| StoreError::DataError(DataError::Empty))?;
             key = format!("{}::{}", new_eav.entity(), new_eav.index());
         }
 
         self.lmdb
             .add(key, &Value::Json(&new_eav.content().to_string()))?;
-        Ok(Some(eav.clone()))
+        Ok(Some(new_eav))
```

Comment:
This seems like an actual logical change. I feel like there should be a test for it if that's the case.

Reply:
Would put the test for sure but to put it in context, this just makes it more consistent than the other eavi in the sense that if there is a collision, a new eavi value should be produced, the lmdb logic was returning an old one but it was not caught because of the times